hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/schemas/InitialModServiceTokenSchema.java

```diff
@@ -604,4 +650,37 @@ private void initializeNetworkRewards(@NonNull final MigrationContext ctx) {
                 .build();
         networkRewardsState.put(networkRewards);
     }
+
+    private void addAdditionalNodesIfAny(
+            @NonNull final WritableKVState<EntityNumber, StakingNodeInfo> stakingInfos,
+            @NonNull final List<NodeInfo> nodeInfos,
+            @NonNull final Configuration config) {
+        final var numberOfNodesInAddressBook = nodeInfos.size();
+        final long maxStakePerNode =
+                config.getConfigData(LedgerConfig.class).totalTinyBarFloat() / numberOfNodesInAddressBook;
+        final var numRewardHistoryStoredPeriods =
+                config.getConfigData(StakingConfig.class).rewardHistoryNumStoredPeriods();
+        final var maxStakeHasChanged = stakingInfos.size() != numberOfNodesInAddressBook;
```

Comment:
I think `stakingInfos.size()` also counts deleted nodes, right? Maybe instead of this flag, we can check if `maxStakePerNode` is different from what is stored.